api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java

```diff
@@ -170,17 +175,74 @@ void shouldSaveClaimantWithClaimStatus(EligibilityStatus eligibilityStatus) {
     }
 
     @Test
-    void shouldSaveDuplicateClaimantForMatchingNino() {
+    void shouldUpdateClaimForMatchingNinoWhenEligible() {
         //given
-        Claimant claimant = aValidClaimant();
-        given(eligibilityAndEntitlementService.evaluateNewClaimant(any()))
-                .willReturn(EligibilityAndEntitlementDecision.buildWithStatus(DUPLICATE));
+        Claimant existingClaimant = aClaimantWithExpectedDeliveryDate(null);
+        LocalDate expectedDeliveryDate = LocalDate.now().plusMonths(6);
+        Claimant newClaimant = aClaimantWithExpectedDeliveryDate(expectedDeliveryDate);
+        Claim existingClaim = aClaimWithClaimant(existingClaimant);
+        UUID existingClaimId = UUID.randomUUID();
+        given(eligibilityAndEntitlementService.evaluateNewClaimant(any())).willReturn(EligibilityAndEntitlementDecision.builder()
+                .eligibilityStatus(ELIGIBLE)
+                .existingClaimId(existingClaimId)
+                .voucherEntitlement(aPaymentCycleVoucherEntitlementWithVouchers())
+                .build());
+        given(claimRepository.findById(any())).willReturn(Optional.of(existingClaim));
 
         //when
-        ClaimResult result = newClaimService.createClaim(claimant);
+        ClaimResult result = newClaimService.createOrUpdateClaim(newClaimant);
 
         //then
-        verify(eligibilityAndEntitlementService).evaluateNewClaimant(claimant);
+        assertThat(result.getClaim()).isEqualTo(existingClaim);
+        assertThat(result.getClaim().getClaimant()).isEqualTo(existingClaimant);
+        assertThat(result.getClaim().getClaimant().getExpectedDeliveryDate()).isEqualTo(expectedDeliveryDate);
+        verify(eligibilityAndEntitlementService).evaluateNewClaimant(newClaimant);
+        verify(claimRepository).findById(existingClaimId);
+        verify(claimRepository).save(result.getClaim());
+        verify(eventAuditor).auditUpdatedClaim(result.getClaim(), singletonList("expectedDeliveryDate"));
+        verifyZeroInteractions(messageQueueDAO);
+    }
+
+    @Test
+    void shouldThrowExceptionWhenExistingClaimNotFound() {
+        //given
+        LocalDate expectedDeliveryDate = LocalDate.now().plusMonths(6);
+        Claimant newClaimant = aClaimantWithExpectedDeliveryDate(expectedDeliveryDate);
+        UUID existingClaimId = UUID.randomUUID();
+        given(eligibilityAndEntitlementService.evaluateNewClaimant(any())).willReturn(EligibilityAndEntitlementDecision.builder()
```

Comment:
Test data factory again?

Reply:
Sticking with builder due to number of fields being set.